accounts-cluster-bench/src/main.rs

```diff
@@ -227,6 +229,264 @@ fn make_close_message(
     Message::new(&instructions, Some(&keypair.pubkey()))
 }
 
+#[derive(Clone, Copy, Debug)]
+pub enum RpcBench {
+    Version,
+    GetSlot,
+    GetMultipleAccounts,
+    GetProgramAccounts,
+    GetTokenAccountsByOwner,
+}
+
+#[derive(Debug)]
+pub enum RpcParseError {
+    InvalidOption,
+}
+
+impl FromStr for RpcBench {
+    type Err = RpcParseError;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "get-slot" => Ok(RpcBench::GetSlot),
+            "get-multiple-accounts" => Ok(RpcBench::GetMultipleAccounts),
+            "get-token-accounts-by-owner" => Ok(RpcBench::GetTokenAccountsByOwner),
+            "version" => Ok(RpcBench::Version),
+            _ => Err(RpcParseError::InvalidOption),
+        }
+    }
+}
+
+fn process_get_multiple_accounts(
+    max_closed: &AtomicU64,
+    max_created: &AtomicU64,
+    stats: &mut RpcBenchStats,
+    last_error: &mut Instant,
+    base_keypair_pubkey: &Pubkey,
+    program_id: &Pubkey,
+    client: &RpcClient,
+) {
+    let start = max_closed.load(Ordering::Relaxed);
+    let end = max_created.load(Ordering::Relaxed);
+    let mut chunk_start = start;
+    let chunk_size = 10;
+    while chunk_start < end {
+        let chunk_end = std::cmp::min(chunk_start + chunk_size, end);
+
+        let addresses: Vec<_> = (chunk_start..chunk_end)
+            .map(|seed| {
+                Pubkey::create_with_seed(base_keypair_pubkey, &seed.to_string(), program_id)
+                    .unwrap()
+            })
+            .collect();
+        chunk_start = chunk_end;
+        let mut rpc_time = Measure::start("rpc-get-multiple-accounts");
+        match client.get_multiple_accounts(&addresses) {
+            Ok(accounts) => {
+                rpc_time.stop();
+                for account in accounts.into_iter().flatten() {
+                    if thread_rng().gen_ratio(1, 10_000) {
+                        info!(
+                            "account: lamports {:?} size: {} owner: {:?}",
+                            account.lamports,
+                            account.data.len(),
+                            account.owner
+                        );
+                    }
+                }
+                stats.total_success_time_us += rpc_time.as_us();
+                stats.success += 1;
+            }
+            Err(e) => {
+                rpc_time.stop();
+                stats.total_errors_time_us += rpc_time.as_us();
+                stats.errors += 1;
+                if last_error.elapsed().as_secs() > 1 {
```

Comment:
Is this intended to only be 1sec? The other elapsed checks are 2sec.

Reply:
changed to 2 to match the others